src/main/java/homework_5/PowerOfNumber/InputValidator.java

```diff
@@ -0,0 +1,55 @@
+package homework_5.PowerOfNumber;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class InputValidator {
+
+    PowerOfNumber powerOfNumber;
+
+    public void run(PowerOfNumber powerOfNumber) {
+        System.out.println("Enter the base and exponent. [a b]");
+        this.powerOfNumber = powerOfNumber;
+        input();
+    }
+
+    private void validate(String[] parameters) {
+
+        if (parameters.length > 2) {
+            errorMsg();
+        }
+
+        try {
+            int a = Integer.parseInt(parameters[0]);
+            int b = Integer.parseInt(parameters[1]);
+            if (a >= 0 && b>= 0){
+                powerOfNumber.input(a, b);
```

Comment:
Validator runs main logic? Well, that is REALLY unexpected. Please fix

Reply:
Expected behaviour of validator is either do nothing if ok or interrupt execution if not ok